### PR TITLE
Update caches systems references to be actual caches and not an additional 0(n) lookup

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
## Overview

The Service registry is effectively a 0n lookup system using a dictionary, the dictionary is used primarily to ensure fast 0n lookup and also provide unique adherence to the single service principle of dependency injection.

This change resolves an issue whereby the current "CoreServices" systems lookups are actually not caching / improving lookup performance by adding an additional 0n lookup on top of / in replacement of the original 0n lookup.

Simply put, this provides direct references to the the instantiated services in the service registry rather than looking them up, potentially every update frame.

I have left in the updated cache table, although it is likely redundant.

## Changes

Simply adds direct reference caches for each of the core systems identified in the "Core Systems" cache.

### CODE REVIEW

File: Assets/MRTK/Core/Utilities/CoreServices.cs [hunk 1]
```diff
@@ -20,59 +20,77 @@ namespace Microsoft.MixedReality.Toolkit
     /// </summary>
     public static class CoreServices
     {
+        private static IMixedRealityBoundarySystem boundarySystem;
```
Comment (keveleigh) [inline Assets/MRTK/Core/Utilities/CoreServices.cs:23]: We investigated this approach a while back (CC @MaxWang-MS) and decided against it, but I don't remember the exact reason now...

If we go this approach now, we'll probably want to clear these new caches in `ResetCacheReference` and `ResetCacheReferences`

Comment (SimonDarksideJ) [reply Assets/MRTK/Core/Utilities/CoreServices.cs:23]: I'll look to update it and clear the caches from that call.  Although I am concerned why you would even need to reset the caches as all the systems listed in this function are transient, meaning they should never be cleared for the lifetime of the runtime.

Comment (SimonDarksideJ) [reply Assets/MRTK/Core/Utilities/CoreServices.cs:23]: Updated as per your request @keveleigh

Comment (keveleigh) [reply Assets/MRTK/Core/Utilities/CoreServices.cs:23]: > all the systems listed in this function are transient, meaning they should never be cleared for the lifetime of the runtime.

Just to make sure we're talking about the same thing, my understanding of "transient" and "should never be cleared for the lifetime of the runtime" are two different things: the first is short-lived, and the other sounds more long-lived.

Either way though, we don't assert that MRTK services are specifically one or the other! Devs may swap profiles at runtime, leading to these cached references potentially going stale. The cache clearing logic all happens in MixedRealityToolkit.cs when the profile is changed and the services are torn down and spun back up.

Comment (keveleigh) [reply Assets/MRTK/Core/Utilities/CoreServices.cs:23]: I'm also going to investigate cherry-picking this to 2.8.1. I think this will help with a selection of perf issues I've been strategically targeting. Thanks!

**Review by keveleigh — APPROVED** [review]

**Comment by keveleigh** [comment]
Thanks for this change!
